@stellar/freighter-api/src/getAddress.ts

```diff
@@ -0,0 +1,20 @@
+import { requestPublicKey } from "@shared/api/external";
+import { FreighterApiError } from "@shared/api/types";
+import { FreighterApiNodeError } from "@shared/api/helpers/extensionMessaging";
+import { isBrowser } from ".";
+
+export const getAddress = async (): Promise<
+  { address: string } | { error: FreighterApiError }
+> => {
+  if (isBrowser) {
+    const req = await requestPublicKey();
+
+    if (req.error) {
+      return { error: req.error };
+    }
+
+    return { address: req.publicKey };
+  }
+
+  return { error: FreighterApiNodeError };
```

Comment:
similar changes are happening to each of the client facing api methods:

If there's an error, return it.

If not, return the requested data.

Also, in the case of a node env, return a "node is not supported" error